### DISCUSSION TITLE
Photo-Realistic Single Image Super-Resolution Using a Generative Adversarial Network

### DISCUSSION THREAD

**Discussion by harshraj22** [harshraj22/DeepLearning#12] [Paper Review]
# SRGaN [:notebook:](https://arxiv.org/abs/1609.04802)
<ins> CVPR 2017 </ins>

## Architectural Details:
The authors use Generative adversarial network to generate the photorealistic high resolution image from their low resolution counter parts.
The Generator network was built using standard Convolutional blocks with skip connections, while the discriminator was also built using standard convolutional blocks.
![Generator](https://github.com/harshraj22/DeepLearning/raw/main/img/srgan_generator.png)
![Discriminator](https://github.com/harshraj22/DeepLearning/raw/main/img/srgan_discriminator.png)

The loss function consisted of two parts:
- Content Loss: 
Compares the ground truth high resolution image with the generated image, and calculate the loss. They use different methods for comparing the images for the same.
    1. MSE Loss: They calculate the pixel wise mean square error between the images for calculating the loss.
    2. VGG Loss: They pass both the images through a pre-trained VGG network, and find the euclidian distance between the feature representations of the two. The claim is that VGG Loss helps in generating more perceptual (ie. photo realistic) image.
![Content Loss](https://github.com/harshraj22/DeepLearning/raw/main/img/srgan_content_loss.png)
 - Adversarial Loss: 
The generator tries to fool the discriminator, and becomes good at generating the realistic images. In this case, they use BinaryCrossEntropy loss for the same as for the discriminator, it is a binary classification problem (separating true data from fake data)
![Overall Loss](https://github.com/harshraj22/DeepLearning/raw/main/img/srgan_overall_loss.png)

They experimentally claim that this Adversarial setting helps them to generate more photo realistic image compared to going for other options like MSE loss.

## Dataset Details:
- The model was trained on randomly sampled 350,000 images from Imagenet dataset.
- The training split of Imagenet contains 1000 categories including humans. [[1](https://datascience.stackexchange.com/a/28857)]
- They crop 96 x 96 image from this random sample, and then perform 4x bicubic downsampling to generate the corresponding Low Resolution Image.
